client/components/Users.js

```diff
@@ -83,12 +45,29 @@ export class Users extends React.Component {
                     {
                     this.state.users.map(user => {
                         return (
-                            <tr key={++id}>
-                                <td>{user}</td>
-                                <td>{`${user}@email.com`}</td>
+                            <tr key={user.id}>
+                                <td>{user.id.slice(0,8)}</td>
+                                <td>{user.email}</td> 
+
+                                {/* Handels Resetting Password */}
                                 <td><button onClick={this.handleResetPassword}>Reset</button></td>
-                                <td><button name={user} value={id} onClick={this.handleDeleteUser}>X</button></td>
-                                <td><input name={user} type="checkbox" onChange={this.handleChecked} /></td>
+
+                                {/* Handels Deleting a User */}
+                                <td><button name={user.name} value={user.id} onClick={() => {
+                                     if(user.isAdmin === false){
+                                         alert('Not Authorized!')
+                                     } else {
+                                         const updatedUsers = this.state.users.filter( u => u.id !== user.id)
+                                         this.setState({users: updatedUsers})
+                                         this.props.deleteUser(user)}
+                                     }
+                                    }>X</button></td>
+
+                                {/* Handels Updating a User to admin/!admin  */}
+                                <td><input name={user} type="checkbox" onChange={(ev) => {
```

Comment:
this checkbox doesn't function how you'd expect. You'd expect it to be checked for admin users and unchecked for everyone else, and when you click it, it changes the status of the user. These are all unchecked and then clicking it changes the status. The authentication for this ability should happen on the backend.

Reply:
I was trying to figure that out. I'll work on that